backend/src/main/java/com/htecgroup/skynest/security/CustomUserDetailsService.java

```diff
@@ -20,6 +21,10 @@ public class CustomUserDetailsService implements UserDetailsService {
   @Override
   public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
     UserDto userDto = userService.findUserByEmail(username);
-    return new User(userDto.getEmail(), userDto.getEncryptedPassword(), new ArrayList<>());
+
+    List<SimpleGrantedAuthority> authorities =
+        List.of(new SimpleGrantedAuthority(userDto.getRole().getName()));
```

Comment:
Since this list has only one item, usually best practice is to use Collections.singletonList

Reply:
I've corrected it.